test/e2e/service_to_service_test.go

```diff
@@ -0,0 +1,126 @@
+// +build e2e
+ 
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+ 
+package e2e
+ 
+import (
+        "fmt"
+        "net/http"
+        "strings"
+        "testing"
+
+        "github.com/knative/serving/test"
+        "github.com/knative/serving/test/spoof"
+
+        "go.uber.org/zap"
+
+        metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+        corev1 "k8s.io/api/core/v1"
+)
+
+var (
+        clients *test.Clients
```

Comment:
In the function createTargetHostEnvVars and sendRequest, I use clients. So I need to declare it here so that I don't need to explicitly pass it to above two functions through arguments.